### DISCUSSION TITLE
retornando string em hexadecimal

### DISCUSSION THREAD

**Discussion by lucianoalien** [OpenSourceCommunityBrasil/REST-DataWare#329] [Q&A]
Primeiramente quero agradecer a todos pelo trabalho e tempo dedicado.
Estou utilizado banco de dados sqlite, RestDataWare na Version : v2.0.8-29(Savage Reign - GitHub) e Delphi Sydney 10.4 community.
Fiz uma api para consultar o banco no servidor que serve para uma aplicação de chat. A parte Mobile que possui um login quando executado para Windows, funciona tudo certo, mas quando executo em Android o retorno das strings para comparação do login vem em hexadecimal, com isso a aplicação não faz o login, aparece os parâmetros como inválidos. Tem algum tratamento que eu precise fazer para os valores das strings virem como caracteres, assim como a copilação para windows ?

**Comment by mobius1qwe** [comment]
como você escreveu que é na parte de login, imagino que o "hexadecimal" seja na senha, mostre por gentileza como você tá fazendo o hash da senha pois parece ser um problema de lib que funciona só com windows

**Reply by lucianoalien** [reply]
procedure TFrmLogin.CarregarUsuarioPrincipal;
begin
   if not Assigned(FrmPrincipal) then
      Application.CreateForm(TFrmPrincipal, FrmPrincipal);

   FrmPrincipal.ID_USUARIO := G_ID_USUARIO;
   FrmPrincipal.ADMIN      := G_ADMIN;
   FrmPrincipal.NOME       := G_NOME;
   FrmPrincipal.TELEFONE   := G_TELEFONE;
   FrmPrincipal.SENHA      := G_SENHA;
   
   // Aqui eu comparo o que me retornou do banco com o que foi digitado no login
   if (G_TELEFONE <> edt_login_telefone.Text) then
   begin
      ShowMessage('Telefone não confere'+CHR(13)+
                  'Fone:'+G_TELEFONE+chr(13)+
                  'Fone Digitado:'+edt_login_telefone.Text);
      Exit;
   end;

   if (G_SENHA <> edt_login_senha.Text) then
   begin
      ShowMessage('Senha não confere'+CHR(13)+
                  'Senha:'+G_SENHA+chr(13)+
                  'Senha Digitado:'+edt_login_senha.Text);
      Exit;
   end;

   FrmPrincipal.BuscarMural('');
   FrmPrincipal.ShowModal;
   FrmLogin.Close;

end;
 a imagem abaixo é copilada no windows, repare as variáveis e seus valores no lado esquerdo

![image](https://user-images.githubusercontent.com/28115846/234108165-c9421094-8884-4709-8f57-46b9bcc1230b.png)

a imagem abaixo é copilada no android 64, repare que os valores das variáveis string estão em hexadecimal, e o valor hexadecimal não é igual ao valor digitado no edit, assim não passa do login

![image](https://user-images.githubusercontent.com/28115846/234109977-7b408b78-3168-4a65-8b66-6a17748fa71a.png)

**Reply by lucianoalien** [reply]
Debugando deu outro erro, segue imagem

![image](https://user-images.githubusercontent.com/28115846/234115185-5f07b124-d537-4f12-8534-458d87cf4cda.png)

não entrou nos IF se forem diferentes os valores

// Aqui eu comparo o que me retornou do banco com o que foi digitado no login
if (G_TELEFONE <> edt_login_telefone.Text) then
begin
ShowMessage('Telefone não confere'+CHR(13)+
'Fone:'+G_TELEFONE+chr(13)+
'Fone Digitado:'+edt_login_telefone.Text);
Exit;
end;

if (G_SENHA <> edt_login_senha.Text) then
begin
ShowMessage('Senha não confere'+CHR(13)+
'Senha:'+G_SENHA+chr(13)+
'Senha Digitado:'+edt_login_senha.Text);
Exit;
end;

o erro é quando chamo o showmodal

**Reply by mobius1qwe** [reply]
não existe modal em mobile. Já o erro aí me parece ter a ver com thread do seu código, tem que conferir se tá vindo corretamente os valores no seu datamodule e se a tabela tá sendo preenchida corretamente. Informações mais aprofundadas ce precisaria pegar suporte de alguém pra debugar o seu código contigo.
